### DISCUSSION TITLE
What happen when a publisher change Name?

### DISCUSSION THREAD

**Discussion by Trouffman** [microsoft/winget-pkgs#130027] [Q&A] [answered]
What would be the correct / recommended process when a publisher change name?

Context :
- PublisherA change name to PublisherB (for any reason like rebranding, acquisiton, sales, splits etc.. - the reason is irrelevant per say)

As this would most likely be a breaking change. I wonder what would be the best way.

1. Create a second package with the new publisher name.
- This will create a duplicate & usually fail the auto-update process.
- Create confusion to have both at the same same, aka : maintain 2 -almost identical- packages.

2. Create a new package on the new name & delete the old name in the PR
- Anything under PublisherA dissapear (I am not aware of notification / message to the users to informa about it)
- Eveyrthing is available right-away under the new PublisherB Name

What would be a good process to do this change?

I think Option 2 make most sense BUT : 
- The tags should be AT LEAST the same as the former PublisherA package (as it is technically the same package)
- Naming of the App/Package should still be similar for users to find it easily
- Tags can be used with the former Publisher Name (aka PublisherA)

This is still pretty rare to happen, but I stumbled upon one of this case this year: 

pre-2022 : Newtek created a tech called NDI, this what later made available on winget as Newtek.NDI5Tools & Newtek.NDI5Runtime.

2022-2023 : Newtek got bought out by "Vizrt Group", but kept its name for a while, then in 2022/2023, NDI became it's own fully independant group within the "Vizrt Group" (aka being disconnected from the Newtek brand).

Since (at least) April 2023 : the Licence Agreement is solely referencing the new entity (Branded "NDI" only), and since Sept. 2023 (at least) the white paper and document published reference exclusively the new publisher Name (aka NDI)

Sept. 2023 : The Newtek brand is being retired and ceased to exist (https://www.vizrt.com/vizrt/press-center/vizrt-unites-portfolio-to-strengthen-customers-and-partners/) 

2023 onwards : The package available should carry the "NDI" Name as a published as the old name is no longer in use / valid.

Posted here for discussion on process as I could not find process/doc about this.

**Comment by Trenly** [comment, marked as answer]
Take a look at our FAQ, which should already answer this question - https://github.com/microsoft/winget-pkgs/blob/master/FAQ.md#what-should-i-do-if-a-package-is-being-published-by-a-new-publisher

The preference is actually to do option #1, just in a bit of a weird way - adding the package in both the old publisher and the new publisher but with some bad data in the old publisher so that after the upgrade, it automatically switches over to the new publisher

**Reply by Trouffman** [reply]
Valid doc link : https://github.com/microsoft/winget-pkgs/blob/master/doc/FAQ.md#what-should-i-do-if-a-package-is-being-published-by-a-new-publisher

Been looking to update this specific app with the proper name (as they had even more changed recently)

**Comment by Trouffman** [comment]
Thanks a bunch! Will look into getting this process to get this resolved. Thanks

**Comment by Trouffman** [comment]
It's been now a decent amount of time since the changes have happened on the following packages :

New (valid package) : 
NDI.NDIRuntime
NDI.NDITools
https://github.com/microsoft/winget-pkgs/tree/master/manifests/n/NDI


Old package : That should now redirect to the New packages above
Newtek.NDI5Runtime
Newtek.NDI5Tools
https://github.com/microsoft/winget-pkgs/tree/master/manifests/n/NewTek


Seem that the proposed good practice would not work as the package name has changed as well.

**Comment by Trouffman** [comment]
Adding my latest comment here for whoever gets here searching.
The process while clear, wasn't straight-forward enough and I ended up simply requesting the removal of the older publisher name.

This technically broke the upgrade path for some users but according to our stats, it was only a limited number.
